cmd/juju/controller/register.go

```diff
@@ -403,36 +525,37 @@ func (c *registerCommand) promptNewPassword(stderr io.Writer, stdin io.Reader) (
 	return password, nil
 }
 
-const errControllerConflicts = `WARNING: You already have a controller registered with the name %q. Please choose a different name for the new controller.
-
-`
-
 func (c *registerCommand) promptControllerName(store jujuclient.ClientStore, suggestedName string, stderr io.Writer, stdin io.Reader) (string, error) {
 	_, err := store.ControllerByName(suggestedName)
 	if err == nil {
-		fmt.Fprintf(stderr, errControllerConflicts, suggestedName)
 		suggestedName = ""
 	}
-	var setMsg string
-	setMsg = "Enter a name for this controller: "
-	if suggestedName != "" {
-		setMsg = fmt.Sprintf("Enter a name for this controller [%s]: ",
-			suggestedName)
-	}
-	fmt.Fprintf(stderr, setMsg)
-	defer stderr.Write([]byte{'\n'})
-	name, err := c.readLine(stdin)
-	if err != nil {
-		return "", errors.Trace(err)
-	}
-	name = strings.TrimSpace(name)
-	if name == "" && suggestedName == "" {
-		return "", errors.NewNotValid(nil, "you must specify a non-empty controller name")
-	}
-	if name == "" && suggestedName != "" {
-		return suggestedName, nil
+	for {
+		var setMsg string
+		setMsg = "Enter a name for this controller: "
+		if suggestedName != "" {
+			setMsg = fmt.Sprintf("Enter a name for this controller [%s]: ", suggestedName)
+		}
+		fmt.Fprintf(stderr, setMsg)
+		name, err := c.readLine(stdin)
+		if err != nil {
+			return "", errors.Trace(err)
+		}
+		name = strings.TrimSpace(name)
+		if name == "" {
+			if suggestedName == "" {
+				fmt.Fprintln(stderr, "You must specify a non-empty controller name.")
+				continue
+			}
+			name = suggestedName
+		}
+		_, err = store.ControllerByName(name)
```

Comment:
I think I prefer to keep the logic simpler - it does no harm to check in all cases.